### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by BertZZ**
A basic comment system. Comments are attached to a post. All of the comment functionality is currently on the index page but I would like to move the new comment form out into its own page and also attach Users to the comments

### CODE REVIEW

File: app/views/posts/index.html.erb
Comment (adamjohnsnow) [inline app/views/posts/index.html.erb]: Do we need a big h2 title "Add a comment:" and a label "body"? Looks a bit funny

File: app/models/comment.rb
Comment (adamjohnsnow) [inline app/models/comment.rb:3]: I think comments should belong to user. At the moment it seems that comments are anonymous. We should aim to have the user who commented visible with the comment.

File: app/controllers/comments_controller.rb
Comment (adamjohnsnow) [inline app/controllers/comments_controller.rb]: Does this need to create post.comment or can it create a comment independently? Doesn't feel like it should be the posts responsibility to create. I could be wrong, maybe ask Kay?

**Review by adamjohnsnow — COMMENTED** [review]
See comments below. It's good, but I think it could be slicker... but who am I to say...

File: app/views/posts/index.html.erb
Comment (BertZZ) [reply app/views/posts/index.html.erb]: No we don't need it

File: app/models/comment.rb
Comment (BertZZ) [reply app/models/comment.rb:3]: This functionality to come in the next version :)

File: app/assets/javascripts/comments.js
Comment (neoeno) [inline app/assets/javascripts/comments.js]: Unused

File: app/assets/stylesheets/comments.scss
Comment (neoeno) [inline app/assets/stylesheets/comments.scss]: Unused

File: app/controllers/comments_controller.rb
Comment (neoeno) [reply app/controllers/comments_controller.rb]: This works. Alternatively:

```ruby
class CommentsController < ApplicationController
  def new
    @comment = Comment.new
  end

  def create
    @post = Comment.create(comment_params)
    redirect_to '/posts'
  end

  private

  def comment_params
    params
      .require(:comment)
      .permit(:body)
      .merge(post_id: params[:post_id])
  end
end
```

File: app/helpers/comments_helper.rb
Comment (neoeno) [inline app/helpers/comments_helper.rb]: Unused

File: app/views/posts/index.html.erb
Comment (neoeno) [inline app/views/posts/index.html.erb]: Indentation

File: app/views/posts/index.html.erb
Comment (neoeno) [inline app/views/posts/index.html.erb]: Indentation

File: config/routes.rb
Comment (neoeno) [inline config/routes.rb]: Nice — good place for it

File: spec/controllers/comments_controller_spec.rb
Comment (neoeno) [inline spec/controllers/comments_controller_spec.rb]: I'd add some tests here or delete

File: spec/features/user_can_comment_on_posts_spec.rb
Comment (neoeno) [inline spec/features/user_can_comment_on_posts_spec.rb]: This file needs another require. Try running it individually with RSpec

**Review by neoeno — CHANGES_REQUESTED** [review]
There's some good work here — but you can do better on the attention to detail! We'll be good to go after these edits.

File: app/assets/javascripts/comments.js
Comment (BertZZ) [reply app/assets/javascripts/comments.js]: Is it bad to commit unused files that rails creates unless you need to use them later or should you always delete them and not commit them??

Comment (neoeno) [reply app/assets/javascripts/comments.js]: I would refer to you to — https://martinfowler.com/bliki/BeckDesignRules.html

My interpretation is:

1. Does it reveal intention? Maybe sometimes, but in these cases not — the files with the actual functionality in them reveal all the useful intention here, and these files communicate nothing about what we might use them for. At the very most they communicate 'a developer added these intending to do... something, that I have no way of knowing'.
2. Since it doesn't, defer to 'fewest elements' and remove them.

It's also worth remembering that adding them back in should we decide we need them is as easy as creating a file! So not a big cost.

File: app/views/posts/index.html.erb
Comment (neoeno) [inline app/views/posts/index.html.erb]: This line should be indented — as it was before

**Review by neoeno — APPROVED** [review]
Just that one indentation and you're good to merge.

**Pull request merged**
